shimmingtoolbox/optimizer/quadprog_optimizer.py

```diff
@@ -0,0 +1,264 @@
+        else:
+            g = np.block([[-np.eye(n), -np.eye(n)], [np.eye(n), -np.eye(n)], [g_bis],
+                          [-np.eye(n), np.zeros([n, n])], [np.eye(n), np.zeros([n, n])],
+                          [np.zeros([n, n]), -np.eye(n)],
+                          [np.zeros([n, n]), np.eye(n)]])
+            # dim(g) = (6n +n_coils, 2n)
+            h = np.block([[np.zeros([n, 1])], [np.zeros([n, 1])], [sum_constraints],
+                          [-lb], [ub], [np.zeros([n, 1])], [np.max(sum_constraints) * np.ones([n, 1])]])
+            # dim(h) = (6n +n_coils, 1)
+        return g, h
+
+    def get_stability_factor(self, coef, unshimmed_vec, coil_mat, factor):
+        """ Objective function to find the stability factor for the quadratic optimization
+
+        Args:
+            coef (np.ndarray): 1D array of channel coefficients
+            unshimmed_vec (np.ndarray): 1D flattened array (point) of the masked unshimmed map
+            coil_mat (np.ndarray): 2D flattened array (point, channel) of masked coils
+                                      (axis 0 must align with unshimmed_vec)
+            factor (float): Devise the result by 'factor'. This allows to scale the output for the minimize function to
+                            avoid positive directional linesearch
+
+        Returns:
+            float: Residuals for least squares optimization
+        """
+        shimmed_vec = unshimmed_vec + coil_mat @ coef
+        return (shimmed_vec).dot(shimmed_vec) / len(unshimmed_vec) / factor + np.abs(coef).dot(self.reg_vector)
+
+    def _get_currents(self, unshimmed_vec, coil_mat, currents_0):
+        """
+        Returns the currents needed for the shimming, redefined from basic_optimizer because of the constraints
+        Args:
+            unshimmed_vec (np.ndarray): 1D flattened array (point) of the masked unshimmed map
+            coil_mat (np.ndarray): 2D flattened array (point, channel) of masked coils
+                                      (axis 0 must align with unshimmed_vec)
+            factor (float): Devise the result by 'factor'. This allows to scale the output for the minimize function to
+                            avoid positive directional linesearch
+            currents_0 (np.ndarray) : Initial guess for the function
+
+        Returns:
+            np.ndarray : Vector of currents that make the better shimming
+
+        """
+        stability_factor = self.get_stability_factor(self._initial_guess_zeros(), unshimmed_vec,
+                                                     np.zeros_like(coil_mat),
+                                                     factor=1)
+        cost_matrix, cost_vector = self.get_cost_matrices(currents_0, unshimmed_vec, coil_mat, stability_factor)
+        ineq_matrix, ineq_vector = self._get_linear_inequality_matrices()
+
+        currents = quadprog.solve_qp(cost_matrix, -cost_vector, ineq_matrix.T, -ineq_vector[:, 0])[0]
+
+        if type(currents) == str:
```

Comment:
You should try to capture the error and raise it. That way we do not get a cryptic error message where we do not know what happened.